lib/spack/docs/tutorial_advanced_packaging.rst

```diff
@@ -0,0 +1,560 @@
+.. _advanced-packaging-tutorial:
+
+============================
+Advanced Topics in Packaging
+============================
+
+While you can quickly accomplish most of the mundane tasks with what
+was covered in :ref:`packaging-tutorial`, there are times when such
+knowledge won't suffice. Usually this happens for libraries that provide
+more than one API and need to let dependents decide which one to use,
+or for packages that provide tools that are invoked at build-time,
+or in other similar situations.
+
+In the following we'll dig into some of the details of package's
+implementation that permit to deal with these rare, but important,
+occurrences. You can rest assured that in every case Spack remains faithful to
+its philosophy: keep simple things simple, but be flexible enough when
+complex requests arise!
+
+-----------------------
+Set up for the tutorial
+-----------------------
+
+The preferred way to follow this tutorial is to use a Docker image:
+
+.. code-block:: console
+
+ $ docker pull alalazo/spack:advanced_packaging_tutorial
+ $ docker run --rm -h advanced-packaging-tutorial -it alalazo/spack:advanced_packaging_tutorial
+ root@advanced-packaging-tutorial:/#
+
+The image comes with Spack and some of the software pre-installed:
+
+.. code-block:: console
+
+  root@advanced-packaging-tutorial:/# spack find
+  ==> 20 installed packages.
+  -- linux-ubuntu16.04-x86_64 / gcc@5.4.0 -------------------------
+  arpack-ng@3.5.0  hdf5@1.10.1   libpciaccess@0.13.5  libtool@2.4.6  m4@1.4.18  ncurses@6.0          openblas@0.2.20  openssl@1.0.2k     superlu@5.2.1       xz@5.2.3
+  cmake@3.9.4      hwloc@1.11.8  libsigsegv@2.11      libxml2@2.9.4  mpich@3.2  netlib-lapack@3.6.1  openmpi@3.0.0    pkg-config@0.29.2  util-macros@1.19.1  zlib@1.2.11
+
+to help focusing on the topic at hand and not on configuration issues.
+If you already started the image, then you can set your preferred editor
```

Comment:
You can set the `EDITOR` environment variable to your preferred editor (vi, emacs, and nano are included in the image).